### DISCUSSION TITLE
Smooth and refine on an simple shape

### DISCUSSION THREAD

**Discussion by Niels-IO** [elalish/manifold#639] [Q&A]
Hi, 
I am currently trying to understand the possibilities of the smooth and refine operations. It seems very powerful!

I am applying the smooth operation on a simple shape:
<img width="586" alt="image" src="https://github.com/elalish/manifold/assets/47059882/98f8e715-29df-48f6-a5a3-969f4c405dc3">

When I set the smoothness for all halfedges to 0, I would expect to get the same 90 degrees angles back as they are in the original geometry.
Instead, with smoothness set to 0 (and refine to 2), I get this shape:
<img width="902" alt="image" src="https://github.com/elalish/manifold/assets/47059882/8e33022c-99fe-4484-b562-7246dc4b519a">
Where there is also an unwanted shift in the vertices:
<img width="641" alt="image" src="https://github.com/elalish/manifold/assets/47059882/61d851fc-ef52-491a-bf50-4eff5a7d0a69">

When I set smoothness to 1, I get a shape that I would expect with this smoothness value:
<img width="881" alt="image" src="https://github.com/elalish/manifold/assets/47059882/16464297-36fa-4455-a0b9-03ad98fc7966">
The vertices also seem to be aligned correctly.

Am I using the sharpendEdges values wrongly?

This is the code:
```javascript


const positions =
  [0,
    0,
    0,
    0,
    0,
    4,
    0,
    4,
    0,
    0,
    4,
    4,
    4,
    0,
    0,
    8,
    0,
    0,
    4,
    0,
    4,
    8,
    0,
    4,
    4,
    4,
    0,
    8,
    4,
    0,
    4,
    4,
    4,
    8,
    4,
    4];

const triangles =
  [0,
    1,
    2,
    6,
    0,
    4,
    4,
    0,
    8,
    1,
    0,
    6,
    10,
    1,
    6,
    0,
    2,
    8,
    8,
    2,
    3,
    2,
    1,
    3,
    3,
    1,
    10,
    8,
    3,
    10,
    7,
    4,
    5,
    4,
    9,
    5,
    4,
    7,
    6,
    7,
    5,
    11,
    11,
    6,
    7,
    8,
    9,
    4,
    9,
    8,
    10,
    11,
    5,
    9,
    6,
    11,
    10,
    9,
    10,
    11]

const triVerts = Uint32Array.from(triangles);
const vertProperties = Float32Array.from(positions);
const cube = new Mesh({ numProp: 3, triVerts, vertProperties });


const sharpend = triangles.map((tri)=>{
 return{ halfedge: tri*3, smoothness:1.0}
})
const sharpend2 = triangles.map((tri)=>{
 return{ halfedge: tri*3+1, smoothness:1.0}
})
const sharpend3 = triangles.map((tri)=>{
 return{ halfedge: tri*3+2, smoothness:1.0}
})

const result = Manifold.smooth(cube, sharpend.concat(sharpend2).concat(sharpend3)).refine(2)

```

**Comment by elalish** [comment]
I *think* you're doing it right, thought all those `concat` calls make it a bit confusing. It's definitely a bug if you set smoothness to zero for all halfedges and you don't get the same volume back. Can you try with some even simpler shapes? Go ahead and file an issue if the problem seems consistent.
